### PR TITLE
[O2B-1051] add sorting to lhc periods page

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- can sort on lhc periods page

Notable changes for developers:
- NA

Changes made to the database:
- NA

### CODE REVIEW

File: lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js [hunk 1]
```diff
@@ -30,6 +31,10 @@ export class LhcPeriodsOverviewModel extends Observable {
         this._pagination.observe(() => this.fetchLhcPeriods());
         this._pagination.itemsPerPageSelector$.observe(() => this.notify());
 
+        this._overviewSortModel = new SortModel();
+        this._overviewSortModel.observe(() => this._applyFilters(true));
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js:35]: I suppose you took it from somewhere, but I think that `sortModel` should not trigger a function called `applyFilters`

File: lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js [hunk 2]
```diff
@@ -45,12 +50,18 @@ export class LhcPeriodsOverviewModel extends Observable {
...
+        };
+
+        const { appliedOn: sortOn, appliedDirection: sortDirection } = this._overviewSortModel;
+        if (sortOn && sortDirection) {
+            queryParams[`sort[${sortOn}]`] = sortDirection;
+        }
 
-        const endpoint = `/api/lhcPeriodsStatistics?${[paramsPhrase].join('&')}`;
+        const paramsPhrase = new URLSearchParams(queryParams).toString();
+        const endpoint = `/api/lhcPeriodsStatistics?${[paramsPhrase]}`;
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js:64]: array not needed

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js (hunk 1 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js]: yes, refactored

File: lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js (hunk 2 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js]: applied

File: lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js [hunk 3]
```diff
@@ -45,12 +53,18 @@ export class LhcPeriodsOverviewModel extends Observable {
...
-        }).toString();
+        };
 
-        const endpoint = `/api/lhcPeriodsStatistics?${[paramsPhrase].join('&')}`;
+        const { appliedOn: sortOn, appliedDirection: sortDirection } = this._overviewSortModel;
+        if (sortOn && sortDirection) {
+            queryParams[`sort[${sortOn}]`] = sortDirection;
+        }
+
+        const paramsPhrase = new URLSearchParams(queryParams).toString();
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js:66]: Please rename it to `queryParametersString` or `queryParametersExpression`

**Review by martinboulais — CHANGES_REQUESTED** [review]

**Review by martinboulais — APPROVED** [review]

**Pull request merged**